echopop/acoustics.py

```diff
@@ -0,0 +1,452 @@
+        Logarithmic acoustic values.
+    """
+    return 10.0 ** (db_values / 10.0)
+
+
+def to_dB(linear_values: Union[np.ndarray, float]) -> np.ndarray:
+    """
+    Converts acoustics values in the linear domain into the logarithmic domain (dB)
+
+    Parameters
+    ----------
+    linear_values: Union[ np.ndarray, float]
+        Linear acoustic values.
+    """
+    return 10 * np.log10(linear_values)
+
+
+def impute_missing_sigma_bs(strata_dict: dict, sigma_bs_stratum: pd.DataFrame) -> pd.DataFrame:
+    """
+    Imputes sigma_bs for strata without measurements or values
+
+    Parameters
+    ----------
+    strata_dict: dict
+        Dictionary containing the complete list of expected stratum values.
+    sigma_bs_stratum: pd.DataFrame
+        Dataframe containing the mean `sigma_bs` calculated for each stratum.
+
+    Notes
+    -----
+    This function iterates through all stratum layers to impute either the
+    nearest neighbor or mean sigma_bs for strata that are missing values.
+    """
+
+    # Collect stratum column name
+    stratum_col = [col for col in sigma_bs_stratum.columns if "stratum" in col.lower()][0]
+
+    # Collect all possible strata
+    strata_options = np.unique(strata_dict["strata_df"][stratum_col])
+
+    # Collect present strata
+    present_strata = np.unique(sigma_bs_stratum[stratum_col]).astype(int)
+
+    # Invert to retrieve missing strata
+    # ---- KS
+    missing_strata = strata_options[~np.isin(strata_options, present_strata)]
+    # -------- Concatenate the existing data with the missing strata
+    sigma_bs_stratum_impute = pd.concat(
+        [
+            sigma_bs_stratum,
+            pd.DataFrame(
+                {
+                    f"{stratum_col}": missing_strata,
+                    "species_id": np.repeat(
+                        np.unique(sigma_bs_stratum.species_id), len(missing_strata)
+                    ),
+                    "sigma_bs_mean": np.repeat(np.nan, len(missing_strata)),
+                }
+            ),
+        ]
+    ).sort_values(stratum_col)
+
+    # Impute values for missing strata
+    # ---- KS
+    if len(missing_strata) > 0:
+        # ---- Find strata intervals to impute over
+        for i in missing_strata:
+            strata_floor = present_strata[present_strata < i]
+            strata_ceil = present_strata[present_strata > i]
+
+            new_stratum_below = np.max(strata_floor) if strata_floor.size > 0 else None
+            new_stratum_above = np.min(strata_ceil) if strata_ceil.size > 0 else None
+
+            sigma_bs_indexed = sigma_bs_stratum_impute[
+                sigma_bs_stratum_impute[stratum_col].isin([new_stratum_below, new_stratum_above])
+            ]
+
+            sigma_bs_stratum_impute.loc[
+                sigma_bs_stratum_impute[stratum_col] == i, "sigma_bs_mean"
+            ] = sigma_bs_indexed["sigma_bs_mean"].mean()
+
+    # Return output
+    return sigma_bs_stratum_impute
+
+
+def summarize_sigma_bs(
```

Comment:
How about renaming this to `calc_mean_sigma_bs`? Summarize does not convey the calculation of mean values.

Reply:
Also, I feel this function would be a lot simpler if you just handle one species only. I am not see error checking if the species in the TS-length relationship is different from what's in the specimen or the reverse, though I also did not try to dig in to see what the edge cases may be.